src/core/Akka.Cluster/ClusterHeartbeat.cs

```diff
@@ -270,37 +290,32 @@ public ExpectedFirstHeartbeat(UniqueAddress @from)
     /// </summary>
     internal sealed class ClusterHeartbeatSenderState
     {
-        public ClusterHeartbeatSenderState(HeartbeatNodeRing ring, ImmutableHashSet<UniqueAddress> unreachable, IFailureDetectorRegistry<Address> failureDetector)
+        public ClusterHeartbeatSenderState(HeartbeatNodeRing ring, ImmutableHashSet<UniqueAddress> oldReceiversNowUnreachable, IFailureDetectorRegistry<Address> failureDetector)
         {
-            FailureDetector = failureDetector;
-            Unreachable = unreachable;
             Ring = ring;
-            ActiveReceivers = Ring.MyReceivers.Value.Union(Unreachable);
+            OldReceiversNowUnreachable = oldReceiversNowUnreachable;
+            FailureDetector = failureDetector;
+            ActiveReceivers = Ring.MyReceivers.Value.Union(OldReceiversNowUnreachable);
         }
 
-        public HeartbeatNodeRing Ring { get; private set; }
+        public HeartbeatNodeRing Ring { get; }
 
-        public ImmutableHashSet<UniqueAddress> Unreachable { get; private set; }
+        public ImmutableHashSet<UniqueAddress> OldReceiversNowUnreachable { get; }
```

Comment:
This will require a rebase - the API approval diff for Akka.Cluster will fire on this line.